components/admin_console/manage_teams_modal/manage_teams_modal.tsx

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import {filterAndSortTeamsByDisplayName} from 'utils/team_utils';
+import * as Utils from 'utils/utils.jsx';
+
+import LoadingScreen from 'components/loading_screen';
+import Avatar from 'components/widgets/users/avatar';
+
+import {Client4} from 'mattermost-redux/client';
+import {isAdmin} from 'mattermost-redux/utils/user_utils';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {ActionResult} from 'mattermost-redux/types/actions';
+import {Team, TeamMembership} from 'mattermost-redux/types/teams';
+
+import RemoveFromTeamButton from './remove_from_team_button';
+import ManageTeamsDropdown from './manage_teams_dropdown';
+
+export type Props = {
+    locale: string;
+    onModalDismissed: () => void;
+    show: boolean;
+    user?: UserProfile;
+    actions: {
+        getTeamMembersForUser: (userId: string) => Promise<ActionResult>;
+        getTeamsForUser: (userId: string) => Promise<ActionResult>;
+        updateTeamMemberSchemeRoles: (teamId: string, userId: string, isSchemeUser: boolean, isSchemeAdmin: boolean) => Promise<ActionResult>;
+        removeUserFromTeam: (teamId: string, userId: string) => Promise<ActionResult>;
+    };
+}
+
+const ManageTeamsModal = ({locale, onModalDismissed, show, user, actions}: Props) => {
+    const [error, setError] = React.useState<JSX.Element | null>(null);
+    const [teams, setTeams] = React.useState<Team[] | null>(null);
+    const [teamMembers, setTeamMembers] = React.useState<TeamMembership[] | null>(null);
+
+    useEffect(() => {
+        if (user) {
+            loadTeamsAndTeamMembers(user);
+        }
+    }, [user]);
```

Comment:
I suppose this is the correct way to replace old `componentDidMount` `UNSAFE_componentWillReceiveProps`.